### PR TITLE
#2313 set dr status endpoint

### LINKED ISSUE #2313: DRC Debugs - Set DR Status Endpoint
### Description

This ticket is to create a backend endpoint to set the status of a design review.

### Acceptance Criteria

- [ ] it should be on the route '/design-reviews/:designReviewId/set-status'
- [ ] it should take in a DesignReviewStatus in the body
- [ ] You should verify that the design review exists and isn't deleted
- [ ] manually test on Postman and include screenshots of the query result

### Proposed Solution

Look at other endpoints for examples

### PR DESCRIPTION
**Pull request by bracyw**
## Changes

Created a endpoint for setting the status of a design review. It can be used by the creator of the design review or app admins. Implemented behavior in design-review, routes, controllers, and services files.

NEW: added unmocked tests

## Test Cases
- change with app admin who is not creator
- set status works when creator is not admin
- fails when user is not admin or creator

tested with postman, results shown below

## Screenshots
before change - with app admin:
Valid set status (with app admin who is not creator) - before
after change - with app admin:
Valid set status with app admin (who is not creator) - after

set status works when creator is not admin
Valid set status (creator who is not admin)

invalid when user is not admin or creator
Invalid user set status

invalid status - produces error
Invalid status - error

## To Do

_Any remaining things that need to get done_

- [ ] Fix variable type issue (described in comments)

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [ ] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #2313

### CODE REVIEW

**Review by superhvarn — CHANGES_REQUESTED** [review]
Looks like some checks are failing. You might need to have more exceptions in the controller method.

File: src/backend/src/controllers/design-reviews.controllers.ts [hunk 1]
```diff
@@ -128,4 +128,19 @@ export default class DesignReviewsController {
...
+  // Set a new status for the design review
+  static async setStatus(req: Request, res: Response, next: NextFunction) {
+    try {
+      const { designReviewId } = req.params;
+      const { status } = req.body;
+      const user = await getCurrentUser(res);
+      const organizationId = getOrganizationId(req.headers);
+
+      const updatedDesignReview = await DesignReviewsService.setStatus(user, designReviewId, status, organizationId);
+      return res.status(200).json(updatedDesignReview);
```
Comment (superhvarn) [inline src/backend/src/controllers/design-reviews.controllers.ts]: I don't think you have to do a return here, same with the error

**Review by superhvarn — CHANGES_REQUESTED** [review]
lgtm, just fix that controller thing

**Review by superhvarn — DISMISSED** [review]
LGTM

File: src/backend/tests/unmocked/design-review.test.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      organization: { organizationId },
+      dr,
+      designReviewId: id
+    } = await createTestDesignReview();
+    orgId = organizationId;
+    designReview = dr;
+    designReviewId = id;
+  });
+
+  // change with app admin who is not creator
```
Comment (walker-sean) [inline src/backend/tests/unmocked/design-review.test.ts]: There is actually a distinction between app admins and simply admins - we are just concerned with admins for this

File: src/backend/tests/unmocked/design-review.test.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import { batmanAppAdmin, financeMember } from '../test-data/users.test-data';
+import DesignReviewsService from '../../src/services/design-reviews.services';
+import { AccessDeniedException } from '../../src/utils/errors.utils';
+import { createTestDesignReview, createTestUser, resetUsers } from '../test-utils';
+import prisma from '../../src/prisma/prisma';
+import { assert } from 'console';
+import { DesignReview, DesignReviewStatus } from 'shared';
+
+describe('Design Reviews', () => {
```
Comment (walker-sean) [inline src/backend/tests/unmocked/design-review.test.ts:10]: These tests are taking on a lot more than they need to, meaning they are becoming very complicated. Take a look at our other unmocked tests and see the structure they fall into, making a call to the endpoint in a specific test case, and checking against what we expect in that case. Anything beyond that is really testing other endpoints or functionality, which is too much overhead for these tests

Comment (walker-sean) [reply src/backend/tests/unmocked/design-review.test.ts:10]: Writing concise, effective tests can be tough to figure out, so let me (or a tech lead) know if you need help and the other examples aren't helping

**Review by walker-sean — CHANGES_REQUESTED** [review]
Nice work! Just some comments on the tests

**Review by walker-sean — APPROVED** [review]

**Pull request merged**